modes/modeUtils.js

```diff
@@ -25,14 +45,37 @@ var ModeUtils = (function() {
         return appInfo.should_show();
       })
       .map(function(app) {
-        return app.get_id();
+        return new switcherApplication.RegularApplication(
+          app.get_id()
+        );
       });
 
+    // get gnome control center panel app ids
+    let gnomeControlCenterAppIDs = gnomeControlCenter.getPanelAppIDs().map(
+      function(appId) {
+        return new switcherApplication.GnomeControlApplication(
+          appId,
+          mainApplicationName
+        );
+      }
+    );
+    // combine Regular Apps ans and Gnome Control Apps ids
+    let allApps = regularAppIDs.concat(gnomeControlCenterAppIDs);
+
+    return allApps;
+  }
+
   let shellAppCache = { lastIndexed: null, apps: [] };
   let shellApps = (force) => {
     const get = () =>
-      appInfos().map(function(appID) {
-        return Shell.AppSystem.get_default().lookup_app(appID);
+      appInfos().map(function(switcherApp) {
+        let shellApp = Shell.AppSystem.get_default().lookup_app(
+          switcherApp.appId
+        );
+        // TODO: should this really be done during appInfos creation?
```

Comment:
I guess it should, and the function should be renamed `getShellApps` or something similar.

Reply:
It would be better to start rendering the UI immediately (especially I would want to have the input field up as soon as possible so we don't lose keyboard input) and update async as computation/filtering/etc finishes. On the web front this is a practically solved problem with the model popularized by React. A deep dive of the model for example at https://overreacted.io/react-as-a-ui-runtime/. I feel the same concepts could be implemented in the narrow case of this UI much more simply than the general case of a runtime for any UI.